src/ops/fuelup_check.rs

```diff
@@ -3,164 +3,168 @@ use crate::{
     commands::check::CheckCommand,
     component::SUPPORTED_PLUGINS,
     config::Config,
+    constants::{CHANNEL_LATEST_FILE_NAME, CHANNEL_NIGHTLY_FILE_NAME},
+    file::read_file,
     fmt::{bold, colored_bold},
+    path::fuelup_dir,
     target_triple::TargetTriple,
-    toolchain::{DistToolchainName, Toolchain},
+    toolchain::{DistToolchainName, OfficialToolchainDescription, Toolchain},
 };
 use anyhow::Result;
 use semver::Version;
-use std::cmp::Ordering::{Equal, Greater, Less};
 use std::collections::HashMap;
 use std::io::Write;
 use std::str::FromStr;
+use std::{
+    cmp::Ordering::{Equal, Greater, Less},
+    path::Path,
+};
+use tempfile::tempdir_in;
 use termcolor::Color;
+use time::Date;
 use tracing::error;
 
 use crate::{component, download::DownloadCfg};
 
-fn collect_versions(channel: Channel) -> HashMap<String, Version> {
-    let mut latest_versions: HashMap<String, Version> = HashMap::new();
+fn collect_versions_and_dates(channel: Channel) -> HashMap<String, (Version, Option<Date>)> {
```

Comment:
```suggestion
fn collect_pkg_versions(channel: Channel) -> HashMap<String, PackageVersion> {
```
^ could this make sense here?